gulpfile.js

```diff
@@ -7,86 +7,116 @@ var gulp            = require('gulp'),
     mocha           = require('gulp-mocha'),
     util            = require('gulp-util'),
     jshint          = require('gulp-jshint'),
-    exec            = require('child_process').exec;
+    exec            = require('child_process').exec,
+    uglify          = require('gulp-uglify'),
+    gutil           = require('gulp-util'),
+    concat          = require('gulp-concat'),
+    ngAnnotate      = require('gulp-ng-annotate'),
+    htmlmin         = require('gulp-html-minifier');
 
 var config = {
-  jshint : ['./*.js', './*/*.js']
+    jshint: ['./*.js', './*/*.js']
 };
 
-gulp.task('clearStart', function(){
+gulp.task('clearStart', function () {
     // Ensure mongod is not running
     exec('killall mongod');
 });
 
 // startup required services to run the app server
-gulp.task('mongod', function() {
+gulp.task('mongod', function () {
     // spawn in a child process mongodb
-    child_process.exec('mongod', function(err,stdout,stderr){
-      if(stderr){
-        console.log('Mongod[Error]: ' + stderr + ' : ' + stdout);
-      } else {
-        console.log(stdout);
-      }
+    child_process.exec('mongod', function (err, stdout, stderr) {
+        if (stderr) {
+            console.log('Mongod[Error]: ' + stderr + ' : ' + stdout);
+        } else {
+            console.log(stdout);
+        }
     });
 });
 
-
 gulp.task('development', function () {
-  nodemon({ 
-    script: 'app.js',
-    ext: 'html js styl',
-    ignore: ['ignored.js'],
-    env: {"NODE_ENV": "development"}
-  })
-  .on('restart', function () {
-    console.log('restarted!');
-  });
+    nodemon({
+            script: 'app.js',
+            ext: 'html js styl',
+            ignore: ['ignored.js'],
+            env: {
+                "NODE_ENV": "development"
+            }
+        })
+        .on('restart', function () {
+            console.log('restarted!');
+        });
 });
 
 gulp.task('runTests', function () {
-    return gulp.src(['test/*.js'], { read: false })
-        .pipe(mocha({ 
+    return gulp.src(['test/*.js'], {
+            read: false
+        })
+        .pipe(mocha({
             reporter: 'spec',
-            timeout:2000
-          }))
-        .on('error', util.log) 
+            timeout: 2000
+        }))
+        .on('error', util.log)
 
-        // exit on end
-        .once('end', function () {
-          process.exit();
-        });
+    // exit on end
+    .once('end', function () {
+        process.exit();
+    });
 });
 
+//moving views from assets to public and minifying them
+gulp.task('viewCompress', function () {
```

Comment:
@atr2982 "viewCompress" is a bit too ambiguous as we have server side and front end views. Also theres another gulp.task named "jsCompress". I'll change this out to "htmlCompress" as it makes a bit more semantic sense, Just wanted to let you know about the change.